extension/cgroupruntimeextension/integration_test.go

```diff
@@ -198,12 +287,25 @@ func TestCgroupV2SudoIntegration(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			// if running in ECS environment, set the ECS metedata URI environment variable
+			// to get the Cgroup CPU quota from the httptest server
+			cleanECS := func() {}
+			if test.setECSMetadataURI {
+				server := testServerECSMetadata(t, test.expectedGoMaxProcs*1024, test.expectedGoMaxProcs*1024)
```

Comment:
Related to the comment above, maybe using `cgroupCpuQuota` might be more appropriate here.

Reply:
The cgroup2 `manager.update()` is still applying the settings. I would even prefer in the ECS tests to put values in `cgroupCpuQuota` and `cgroupCpuPeriod` that have nothing to do with the `expectedGoMaxProcs` value. To prove the code fetched the value from the HTTP endpoint.

Because if `IsECS() == true` whatever you set in the cgroup, the lib will make a http call to the ECS metadata uri and set `GOMAXPROCS`.

So I see 3 viable options:

- Keep as is
- Only change the `cgroupCpuQuota`.. not aligned with `expectedGoMaxProcs` to prove the HTTP call is happening and working
- Move the ECS tests into a new func next to `func TestCgroupV2SudoIntegration(t *testing.T) {}`

Reply:
Taking into account that `cgroupCpuQuota` and `cgroupCpuPeriod` won't be taken into consideration if `IsECS() == true`, I would go with option 3. It seems to me that having a new `TestECSCgroupV2SudoIntegration` will provide a cleaner testing interface.